### PR TITLE
Refactor WinRM Ctrl+C Signal Handling in CtrlC class

### PR DESCRIPTION
**Pull request by popeye133**
- Consolidated XML handling using Nokogiri::XML::Builder for improved readability and performance.
- Enhanced error handling in handle_response method to raise specific WinRMError on non-200 responses.
- Updated header construction to use array concatenation for efficiency and clarity.
- Introduced stricter validation in initialize method to ensure mandatory options are provided.
- Documented usage and added namespace constants for better code understanding and maintainability.

Tell us what this change does. If you're fixing a bug, please mention
the github issue number.

Please ensure you are submitting **from a unique branch** in your [repository](https://github.com/rapid7/metasploit-framework/pull/11086#issuecomment-445506416) to master in Rapid7's.

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `use exploit/windows/smb/ms08_067_netapi`
- [x] ...
- [x] **Verify** the thing does what it should
- [x] **Verify** the thing does not do what it should not
- [x] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

If you are opening a PR for a new module that exploits a **specific** piece of hardware or requires a **complex or hard-to-find** testing environment, we recommend that you send us a demo of your module executing correctly. Seeing your module in action will help us review your PR faster!

Specific Hardware Examples:
* Switches
* Routers
* IP Cameras
* IoT devices

Complex Software Examples:
* Expensive proprietary software
* Software with an extensive installation process
* Software that requires exploit testing across multiple significantly different versions
* Software without an English language UI

We will also accept demonstrations of successful module execution even if your module doesn't meet the above conditions. It's not a necessity, but it may help us land your module faster!

Demonstration of successful module execution can take the form of a packet capture (pcap) or a screen recording. You can send pcaps and recordings to [msfdev@metasploit.com](mailto:msfdev@metasploit.com). Please include a CVE number in the subject header (if applicable), and a link to your PR in the email body.
If you wish to sanitize your pcap, please see the [wiki](https://docs.metasploit.com/docs/development/get-started/sanitizing-pcaps.html).

### CODE REVIEW

**Comment by adfoster-r7** [comment]
Thanks for the pull request! In terms of verification steps, what would you recommend running through to test the changes?

**Comment by dledda-r7** [comment]
It also looks like the URL `http://schemas.microsoft.com/wbem/wsman/1/windows/shell/signal/ctrl_c` doesn't exist anymore. Not related to the PRs changes itself but it can be challenging to test it now.

File: lib/net/winrm/ctrl_c.rb [hunk 1]
```diff
@@ -1,47 +1,84 @@
...
       end
 
-      protected
+      # Sends the Ctrl+C signal
+      #
+      # @param async [Boolean] Whether to send signal asynchronously (default: false)
+      # @param timeout [Integer] Timeout in seconds for the operation (default: nil)
+      #
+      # @return [Boolean] True if signal sent successfully, false otherwise
+      def send_signal(async: false, timeout: nil)
```
Comment (cdelafuente-r7) [inline lib/net/winrm/ctrl_c.rb:31]: I cannot find where `send_signal` is called from. I might be missing something. Please, can you elaborate?

File: lib/net/winrm/ctrl_c.rb [hunk 2]
```diff
@@ -1,47 +1,84 @@
...
+        request = build_request
+        response = send_request(request, async: async, timeout: timeout)
+        handle_response(response)
+      end
+
+      private
 
-      def create_header(header)
```
Comment (cdelafuente-r7) [inline lib/net/winrm/ctrl_c.rb:21]: It looks like `create_header` is called from `build` in the parent class.

https://github.com/WinRb/WinRM/blob/167ec0600224aced933535e9f2a6ef702141c9aa/lib/winrm/wsmv/base.rb#L29-L40

Wouldn't it be an issue if it is removed?

File: lib/net/winrm/ctrl_c.rb [hunk 3]
```diff
@@ -1,47 +1,84 @@
...
-        header << Gyoku.xml(ctrl_c_header)
+      # Validates the options passed during initialization
+      def validate_options(opts)
+        raise ArgumentError, 'opts[:shell_id] is required' unless opts[:shell_id]
+        raise ArgumentError, 'opts[:command_id] is required' unless opts[:command_id]
       end
 
-      def create_body(body)
```
Comment (cdelafuente-r7) [inline lib/net/winrm/ctrl_c.rb:25]: Same question here. `create_header` seems to be called from `build` in the parent class.

File: lib/net/winrm/ctrl_c.rb [hunk 4]
```diff
@@ -1,47 +1,84 @@
...
-          "#{NS_WIN_SHELL}:Code" =>
-            'http://schemas.microsoft.com/wbem/wsman/1/windows/shell/signal/ctrl_c'
-        }
+      # Handles the response from WinRM server
+      #
+      # @param response [Object] Response object from WinRM server
+      #
+      # @raise [WinRMError] If response indicates failure
+      #
+      # @return [Boolean] True if response indicates success, false otherwise
```
Comment (cdelafuente-r7) [inline lib/net/winrm/ctrl_c.rb:77]: It looks like this method only returns `true` or raises an exception.

**Comment by smcintyre-r7** [comment]
I'm going to attic this because there hasn't been a response to any of the feedback, or the question about what steps should be taken for testing.

**Pull request closed without merging**